### ISSUE TITLE
Live Stream Challenges

### ISSUE THREAD

**Issue by j6s33m** [tsightler/ring-mqtt#253]
I am wondering if anyone else is having this problem. I run both ring-mqtt and home assistant as docker containers on my synology. I have ssl enabled for home assistant with a duckdns domain. I've followed the directions very carefully on how to add livestream, but for some reason, I just cannot get the livestream feature to work on any of my ring cameras. Out of curiosity, @tsightler , were you able to test your model using docker containers w/ HA vs it just as a HA add-on?

**Comment by j6s33m** [comment]
I figured it out! In the livestream URL, my HA wanted to put the container name "ring-mqtt:8554" and instead, I changed the container name to the container url with port and it worked. Can't believe I missed that. 

Thanks again @tsightler

**Comment by tsightler** [comment]
Glad you figured it out.  Can you clarify what you had to change as perhaps I can improve the logic that attempts to "guess" the livestream URL.

**Comment by j6s33m** [comment]
Sure. In stream source, my camera's info shows this:

rtsp://[username]:[password]@**ring-mqtt:8554**/[cameraID]_live

with "ring-mqtt" being the name of my docker container, but you need to change "ring-mqtt" to be the url of your docker container. In my case, once I changed that to be "192.168.xxx.xxx:8544", it worked.

**Comment by j6s33m** [comment]
I haven't been able to get the still images to work yet so I'm trying to figure out that one too.

**Comment by tsightler** [comment]
When you say URL of your Docker, do you just mean IP address?  I'm not aware of the Docker container having a URL, but hey, I could be wrong.  I knew that the streaming URL would likely be incorrect for Docker installs because I have no way to determine a valid hostname or even if it's possible, maybe I should just use the hostname when running as the addon as the IP for docker installs, but then it will be wrong if HA and ring-mqtt run on different hosts, but that's probably not too common.

There's actually a typo in the still image URL attributes.entity_picture vs attribute.entity_picture.  Also, in the next release I plan to query the HA API to get the external URL because, for example, if you are using SSL, then using local host will not work unless you also disbale verify_ssl.

**Comment by j6s33m** [comment]
Yes...I apologize. I mean IP. 

I'll try to adjust the typo and see if that fixes my issue now.

**Comment by j6s33m** [comment]
Ya I still can't figure out how to get the entity picture. Here's what shows up in my developer tools:

entity_picture: /api/camera_proxy/camera.driveway_live?token=[alphanumeric code here]

My config:

- mqtt-ring = docker container
- homeassistant = docker container
- SSL for HASS via duckdns

**Comment by tsightler** [comment]
That looks like you are trying to get the still image from the live camera proxy, that is unlikely to work as the live camera does not run all of the time.  You have to enable snapshots and use the camera proxy on the snapshot camera entity.

If you do that, it should be something like:

https://<your_normal_ha_url>:8123{{ states.camera.driveway_snapshot.attributes.entity_picture }}

If you've change the default port from 8123 to something else (for example, I set mine to the standard default SSL port of 443) then you'll need to modify that portion of the URL to match whatever URL you normally use to access your HA instance.

Basically, it's simply querying itself on it's normal URL, and using the value from the entity_picture attribute on the given camera to complete the URL.  If you copy and past that entity_picture URL, and tack it on to your normal HA URL, you can see the picture in your browser.

**Comment by j6s33m** [comment]
Ok that makes sense. What's odd is that I can't find snapshots in my entity list. I have snapshotmode in my container environment, but doesn't seem to create snapshot entities.

<img width="435" alt="Screen Shot 2021-09-08 at 2 59 37 PM" src="https://user-images.githubusercontent.com/17929290/132577039-bbc828d0-8217-4f1e-9cfd-3ef12b95adc2.png">

**Comment by tsightler** [comment]
The snapshot feature has been around a long time and is well proven, I'm not sure how to help you if you can find the snapshot camera entities.  Do you see the camera devices in Home Assistant?  Not the live cameras you manually created, but the actual MQTT camera devices with motion sensors, stream switches, etc?  And in there you don't see any a snapshot camera entity?

**Comment by j6s33m** [comment]
Yes I have the MQTT camera devices ... see below, but have never seen the snapshot camera entity. Up until today, I also leveraged the native ring cloud integration and was able to get recent motion snapshots through that, but have never been able to get them through the MQTT integration. 

PS - You really need a "buy me a coffee" link.

<img width="359" alt="Screen Shot 2021-09-08 at 3 45 55 PM" src="https://user-images.githubusercontent.com/17929290/132583052-67249b76-1e81-4c5d-98c3-245589574fc2.png">

**Comment by tsightler** [comment]
I don't have any clues on why the snapshot entities don't show up, like I said it's a feature that's been around a long time and there are no known issue with it.  It's widely used and people aren't usually shy about opening issues if something is not working with them.

Did you only recently enabled them?  If so make sure to perform a browser refresh while on the page, sometimes HA doesn't display new entities automatically with some weird caching issue.  If you can send me the logs I can check to see if the discovery messages are in there so we know whether it's either a) not being enabled in the addon properly, or b) not being recognized by HA.  That would at least help to narrow down where the problem is.  I mean, it's really weird because you also don't have the number entity for the snapshot interval either, so that implies that it's not enabled properly, but your screenshot looked correct, although it's a very limited view.  Are you sure you completely stopped and started a new ring-mqtt container with the new environment variables?

**Comment by j6s33m** [comment]
So it works now. I removed the single quotes around the word **all** in my environment variables. I always had quotes there because your writeup below shows single quotes around the word **disabled** ... Removing the quotes fixed the problem. Everything is working....sorry to be such a pain. Really appreciate all of the work you do on this project. Please create a "buy me a coffee" so that happy users like myself can say thank you.

<img width="811" alt="Screen Shot 2021-09-08 at 4 15 54 PM" src="https://user-images.githubusercontent.com/17929290/132586744-1a7bc40e-ccc2-4baa-a24b-d904b4fbfc5b.png">

**Comment by tsightler** [comment]
The writeup isn't assuming you're using some fancy GUI that takes of of quotes and things for you!  :-)

Glad you got it working.  I don't bother with a coffee link because this is mostly an educational experience for me and I have a great job that treats and pays me well so I'd feel weird taking people's money, even if they are volunteering it.  If you feel like some money should go somewhere, find a worthy cause and/or person and give it to them as I certainly don't feel like I'm that person.  Otherwise, just a simple thanks is totally fine.

I wish there was a "donate to a cause" button instead of a "buy me a coffee" button.  Maybe there is one and I just don't know about it.

**Comment by j6s33m** [comment]
Cheers and thanks again!
